base/file.jl

```diff
@@ -230,23 +230,27 @@ julia> mkpath("intermediate_dir/actually_a_directory.txt") # creates two directo
 julia> isdir("intermediate_dir/actually_a_directory.txt")
 true
 
+julia> mkpath("my/test/dir/") # returns the original `path`
+"my/test/dir/"
 ```
 """
 function mkpath(path::AbstractString; mode::Integer = 0o777)
-    isdirpath(path) && (path = dirname(path))
     dir = dirname(path)
+    # stop recursion for `""`, `"/"`, or existed dir
     (path == dir || isdir(path)) && return path
     mkpath(dir, mode = checkmode(mode))
     try
-        mkdir(path, mode = mode)
+        # cases like `mkpath("x/")` will cause an error if `isdir(path)` is skipped
```

Comment:
Putting myself again into the shoes of someone reading this in the future without this context, I believe I would be quite confused. Why would "isdir(path)" be skipped?!?

even now I am not entirely sure what you are trying to say. Maybe: "create the directory if it doesn't already exist"?